cli/azd/pkg/exec/runargs.go

```diff
@@ -42,6 +44,16 @@ func NewRunArgs(cmd string, args ...string) RunArgs {
 	}
 }
 
+// NewRunArgs creates a new instance with the specified cmd and args and a list of SensitiveData
+// Use this constructor to protect known sensitive data from going to logs
+func NewRunArgsWithSensitiveData(cmd string, args, sensitiveData []string) RunArgs {
```

Comment:
This new constructor is to allow protecting information within the arguments which might not have a regex, like a PAT.

Example:

![image](https://user-images.githubusercontent.com/24213737/236146021-88931a44-19f5-4192-a000-a1f973dbcff6.png)

Reply:
Could imagine in the future we expand this such that the caller can pass in a series of regular expressions if they need a bit more power.  I do like allowing the caller the ability to tell the system "here's the sensitive stuff for me in the context of this command", thanks for adding this!

Reply:
Updating the logs example when using `-c foo=bar`: 

![image](https://user-images.githubusercontent.com/24213737/236342633-a9d2368c-2e09-4ee6-98df-2bec08681e8b.png)